src/NUnitEngine/nunit.engine/Services/AgentProcess.cs

```diff
@@ -88,16 +96,34 @@ public static string GetTestAgentExePath(RuntimeFramework targetRuntime, bool re
                 // bit of a kluge, but it's necessary unless we change the binary 
                 // output directory to match the distribution structure.
                 agentsDir = Path.Combine(Path.GetDirectoryName(engineDir), "agents");
-                log.Debug($"Directory not found! Using {agentsDir}");
+                log.Debug("Assuming test run in project output directory");
             }
 
-            string runtimeDir = targetRuntime.FrameworkVersion.Major >= 4 ? "net40" : "net20";
+            log.Debug($"Checking for agents at {agentsDir}");
 
             string agentName = requires32Bit
-                ? "nunit-agent-x86.exe"
-                : "nunit-agent.exe";
+                ? "nunit-agent-x86"
+                : "nunit-agent";
+
+            string runtimeDir;
+            string agentExtension;
+            switch (targetRuntime.Runtime)
+            {
+                case RuntimeType.Net:
+                case RuntimeType.Mono:
+                    runtimeDir = targetRuntime.FrameworkVersion.Major >= 4 ? "net40" : "net20";
+                    agentExtension = ".exe";
+                    break;
+                case RuntimeType.NetCore:
+                    runtimeDir = "netcoreapp3.1";
+                    agentExtension = ".dll";
+                    break;
+                default:
+                    log.Error($"Unknown runtime type: {targetRuntime.Runtime}");
+                    return null;
+            }
```

Comment:
I think this code needs to match a decision on an netcoreapp31 x86 agent, right? Right now, if I ran a .NET Core 3.1 tests with the --x86 flag, I think we'd end up with an agent exe not being found?

Reply:
Yes... I didn't come back and change this after I discovered we didn't need an x86 agent for .NET Core. I'll need to change TestAgency as well.

Reply:
Done, but continues to incorporate a hard-coded path. We'll need to revisit for dotnet installations at a non-standard location.